### PR TITLE
DEV-316: Add build tags to Mac launcher

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20808/HighFidelity-Beta-PR16046-d5e748e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20808/HighFidelity-Beta-Interface-PR16046-d5e748e.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20808/HighFidelity-Beta-PR16046-d5e748e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20808/HighFidelity-Quest-Beta-PR16046-d5e748e.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20808/HighFidelity-Beta-PR16046-d5e748e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20808/HighFidelity-Beta-Interface-PR16046-d5e748e.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20896/HighFidelity-Beta-PR16046-913c4b0.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20896/HighFidelity-Quest-Beta-PR16046-913c4b0.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20896/HighFidelity-Beta-PR16046-913c4b0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20896/HighFidelity-Beta-Interface-PR16046-913c4b0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20896/HighFidelity-Beta-PR16046-913c4b0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20896/HighFidelity-Beta-Interface-PR16046-913c4b0.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20935/HighFidelity-Beta-PR16046-73c120b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20935/HighFidelity-Beta-Interface-PR16046-73c120b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20935/HighFidelity-Beta-PR16046-73c120b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20935/HighFidelity-Beta-Interface-PR16046-73c120b.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20935/HighFidelity-Beta-PR16046-73c120b.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20935/HighFidelity-Quest-Beta-PR16046-73c120b.apk)

File: launchers/darwin/src/Launcher.m [hunk 1]
```diff
@@ -275,13 +281,27 @@ - (void) displayNameEntered:(NSString*)aDiplayName
...
+        NSError* error = nil;
+        Interface* interface = [[Interface alloc] initWith:interfaceAppPath];
+        currentVersion = [interface getVersion:&error];
+        if (currentVersion == 0 && error != nil) {
+            NSLog(@"can't get version from interface, falling back to settings: %@", error);
+            currentVersion = [Settings.sharedSettings latestBuildVersion];
+        }
+    } @catch (NSException *exception) {
+        NSLog(@"an exception was thrown: %@", exception);
+        currentVersion = [Settings.sharedSettings latestBuildVersion];
```
Comment (danteruiz) [inline launchers/darwin/src/Launcher.m]: If there is not interface.app and currentVersion is the same as the latestBuildRequest. We will not download a new interface

File: launchers/darwin/src/Launcher.m [hunk 2]
```diff
@@ -349,49 +370,94 @@ - (BOOL)canBecomeKeyWindow
...
+    }
+
+    for (NSDictionary* build in self.latestBuilds) {
+        NSString* name = [build valueForKey:@"name"];
+        NSLog(@"Checking %@", name);
+        if ([name isEqual:buildTag]) {
+            url = [[[build objectForKey:@"installers"] objectForKey:@"mac"] valueForKey:@"zip_url"];
+            NSString* thisLatestVersion = [build valueForKey:@"latest_version"];
+            latestVersion = thisLatestVersion.integerValue;
+            newVersionAvailable = currentVersion != latestVersion;
```
Comment (danteruiz) [inline launchers/darwin/src/Launcher.m]: look at my comment above.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21009/HighFidelity-Beta-PR16046-1eaba67.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21009/HighFidelity-Quest-Beta-PR16046-1eaba67.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21009/HighFidelity-Beta-PR16046-1eaba67.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21009/HighFidelity-Beta-Interface-PR16046-1eaba67.dmg

**Comment by shanzzam** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21012/HighFidelity-Beta-PR16046-1eaba67.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21012/HighFidelity-Beta-Interface-PR16046-1eaba67.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21012/HighFidelity-Beta-PR16046-1eaba67.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21012/HighFidelity-Beta-Interface-PR16046-1eaba67.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21012/HighFidelity-Beta-PR16046-1eaba67.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21012/HighFidelity-Quest-Beta-PR16046-1eaba67.apk)

**Pull request merged**